docs/control-plane-architecture.adoc

```diff
@@ -0,0 +1,252 @@
+** a **bootstore** containing basic network configuration needed to bring up the rack
+** information about what control plane services are running on this sled
+* Sled agents apply any needed network configuration and start any services they're supposed to be running:
+** On Scrimlets, the switch zone and boundary NTP are started.  Boundary NTP synchronizes time from the customer-provided NTP servers.
+** On non-Scrimlets, internal DNS is started.  The rest of cold boot waits until time has been synchronized from the boundary NTP instances.
+** Once time is synchronized, internal DNS services are started so that components can find each other.
+** Once internal DNS is available, all other services are started concurrently.
+*** CockroachDB nodes start up, discover the rest of the cluster via DNS, and form a cluster.
+*** Nexus starts up and waits for CockroachDB to become available.
+*** All other services start up and wait for their dependencies to become available.
+
+For this to work:
+
+* **Bootstore** must contain enough information to configure networking on the switches and each host to reach other services within the rack as well as the outside world (for NTP).
+* **Internal DNS** must be able to come up without any external dependencies, meaning it stores a complete copy of all DNS data locally.
+
+However, Nexus is the place where all _changes_ to configuration are made, and CockroachDB is the source of truth for all configuration.  As a result, when changing bootstore contents or internal DNS, the change is first made at Nexus, stored into CockroachDB, and then propagated to all sleds and internal DNS instances for local persistent storage so that it's available on cold start (of the _sled_) without the rest of the control plane being up.
+
+This is a very rough approximation, but gives an idea of the dependencies associated with cold start.
+
+[bibliography]
+== References
+
+Unfortunately, most of these RFDs are not yet public.
+
+* [[[rfd4, RFD 4]]] https://4.rfd.oxide.computer/[RFD 4 User Facing API]
+* [[[rfd6, RFD 6]]] https://6.rfd.oxide.computer/[RFD 6 Threat Model]
+* [[[rfd24, RFD 24]]] https://24.rfd.oxide.computer/[RFD 24 Multi-Rack Oxide Deployments]
+* [[[rfd30, RFD 30]]] https://30.rfd.oxide.computer/[RFD 30 Oxide Console Prototype]
+* [[[rfd48, RFD 48]]] https://48.rfd.oxide.computer/[RFD 48 Control Plane Requirements]
+* [[[rfd53, RFD 53]]] https://53.rfd.oxide.computer/[RFD 53 Control plane data storage requirements]
+* [[[rfd61, RFD 61]]] https://61.rfd.oxide.computer/[RFD 61 Control Plane Architecture]
+* [[[rfd110, RFD 110]]] https://110.rfd.oxide.computer/[RFD 110 CockroachDB for the control plane database]
+* [[[rfd125, RFD 125]]] https://125.rfd.oxide.computer/[RFD 125 Telemetry requirements and building blocks]
+* [[[rfd162, RFD 162]]] https://162.rfd.oxide.computer/[RFD 162 Metrics collection architecture and design]
```

Comment:
This link didn't work for me for some reason.

Reply:
It gets redirected to https://github.com/oxidecomputer/rfd/tree/0162/rfd/0162/, which doesn't work because that RFD was published and it should therefore go to master, i.e., https://github.com/oxidecomputer/rfd/tree/master/rfd/0162/.

These redirects are powered by an old service we haven't touched in a long time. I just talked to @augustuswm about fixing it so these go to `rfd.shared.oxide.computer/rfd/0000` instead. He's going to take care of it.